File: Exercise/LuoGu/Code_2023_7_30/P1423.java
package Code_2023_7_30;

import java.util.Scanner;

public class P1423 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        double distance = sc.nextDouble();
        double sumDistance = 0, step = 2;
        int stepCnt = 0;

        while (sumDistance < distance) {
            stepCnt++;
            sumDistance += step;
            step *= 0.98;
        }

        System.out.println(stepCnt);

        sc.close();
    }
}

File: Exercise/LuoGu/Code_2023_7_30/P1423.cpp
#include <iostream>
#include <algorithm>
#include <math.h>
#include <vector>
#include <string>

using namespace std;

int main(void)
{
    double distance, step = 2, sum_distance = 0;
    int step_cnt = 0;
    cin >> distance;
    while (sum_distance < distance)
    {
        step_cnt++;
        sum_distance += step;
        step *= 0.98;
    }
    cout << step_cnt << endl;
    return 0;
}